File: benhussein/report/gap_analysis_report/gap_analysis_report.py
"""GAP Analysis Report."""

import frappe


def execute(filters=None):
    columns = [
        {"fieldname": "metal", "label": "Metal", "fieldtype": "Data", "width": 120},
        {"fieldname": "gap", "label": "Gap (%)", "fieldtype": "Percent", "width": 120},
    ]
    data = []
    return columns, data

File: benhussein/report/gap_analysis_report/gap_analysis_report.js
frappe.query_reports["GAP Analysis Report"] = {
  filters: [
    {
      fieldname: "from_date",
      label: __("From"),
      fieldtype: "Date",
    },
    {
      fieldname: "to_date",
      label: __("To"),
      fieldtype: "Date",
    },
  ],
};
